en/guide/extending_csd.md

```diff
@@ -1,6 +1,121 @@
+
+```cpp
+class CameraParameters {
+```
+#### 1. Extend CameraDevice Class
+To add support for new type of camera device in CSD, a custom class need to be derived from CameraDevice base class.
+
+Example: CameraDeviceGazebo is the class that extends CameraDevice class.
+
+```cpp
+class CameraDeviceGazebo final : public CameraDevice {
+```
+CameraDevice has some pure virtual functions (runtime binding) that must be implemented by the child class. Other methods can also be overloaded to provide the functionality.
+
+Example : Details on Gazebo class
+
+```cpp
+class CameraDeviceGazebo final : public CameraDevice {
+public:
+    CameraDeviceGazebo(std::string device);
+    ~CameraDeviceGazebo();
+    std::string getDeviceId();
+    int getInfo(struct CameraInfo &camInfo);
+    bool isGstV4l2Src();
+    int init(CameraParameters &camParam);
+    int uninit();
+    int resetParams(CameraParameters &camParam);
+    int setParam(CameraParameters &camParam, std::string param, const char *param_value,
+                 size_t value_size, int param_type);
+```
+#### 2. Detect Camera Device
+After having added a custom CameraDevice class, there has be a logic in place to detect the custom camera device. A CameraComponent will be created for every camera device detected. V4L2 camera devices are detected by scanning the linux device nodes /dev/video*. Gazebo camera is detected based on --enable-gazebo compile time flag. Configurations associated with the camera device is read from the conf file.
+
+```cpp
+// prepare the list of cameras in the system
+int CameraServer::detectCamera(ConfFile &conf)
+```
+Example: For Gazebo CameraDevice
+
+```cpp
+int CameraServer::detect_devices_gazebo(ConfFile &conf, std::vector<CameraComponent *> &camList)
+```
+
+#### 3. Instantiate Camera Device
+The CameraComponent instantiates and owns the custom CameraDevice object. The instantiation of the object is based on the string that identifies the camera device.
+
+```cpp
+std::shared_ptr<CameraDevice> CameraComponent::create_camera_device(std::string camdev_name)
+```
+Example: For Gazebo CameraDevice
+
+```cpp
+    } else if (camdev_name.find("camera/image") != std::string::npos) {
+        log_debug("Gazebo device : %s", camdev_name.c_str());
+#ifdef ENABLE_GAZEBO
+        return std::make_shared<CameraDeviceGazebo>(camdev_name);
+```
+
+#### 4. Declare Camera Parameters
+Parameters supported by the CameraDevice need to be declared by the custom CameraDevice. Either pre-defined parameter strings and IDs can be used or newer ones can be defined. Setting or these parameters and resetting of all the parameters need to be handled by the custom CameraDevice.
```

Comment:
This is part of CameraDevice docs, so should be up near them - not separated by other topics. You mention the parameters above, so probably back link to them.

Reply:
Adding new parameters is something that the developers must do to support new parameters for `CameraDeviceCustom` . 
13. Phrase the terms appropriately, add link to the detailed description of parameters